src/cdk/datepicker/datepicker-input.ts

```diff
@@ -0,0 +1,361 @@
+  }
+  _dateFilter: (date: D | null) => boolean;
+
+  /** The value of the input. */
+  @Input()
+  get value(): D | null { return this._value; }
+  set value(value: D | null) {
+    value = this._dateAdapter.deserialize(value);
+    this._lastValueValid = !value || this._dateAdapter.isValid(value);
+    value = this._getValidDateOrNull(value);
+    const oldDate = this.value;
+    this._value = value;
+    this.emitValue(oldDate, value);
+
+  }
+  protected _value: D | null;
+
+  /** Emit value change if dates are different. */
+  emitValue(oldDate: D| null, value: D| null) {
+    if (!this._dateAdapter.sameDate(oldDate, value)) {
+      this._valueChange.emit(value);
+    }
+  }
+
+  /** The minimum valid date. */
+  @Input()
+  get min(): D | null { return this._min; }
+  set min(value: D | null) {
+    this._min = this._getValidDateOrNull(this._dateAdapter.deserialize(value));
+    this._validatorOnChange();
+  }
+  private _min: D | null;
+
+  /** The maximum valid date. */
+  @Input()
+  get max(): D | null { return this._max; }
+  set max(value: D | null) {
+    this._max = this._getValidDateOrNull(this._dateAdapter.deserialize(value));
+    this._validatorOnChange();
+  }
+  private _max: D | null;
+
+  /** Whether the datepicker-input is disabled. */
+  @Input()
+  get disabled(): boolean { return !!this._disabled; }
+  set disabled(value: boolean) {
+    const newValue = coerceBooleanProperty(value);
+    const element = this._elementRef.nativeElement;
+
+    if (this._disabled !== newValue) {
+      this._disabled = newValue;
+      this._disabledChange.emit(newValue);
+    }
+
+    // We need to null check the `blur` method, because it's undefined during SSR.
+    if (newValue && element.blur) {
+      // Normally, native input elements automatically blur if they turn disabled. This behavior
+      // is problematic, because it would mean that it triggers another change detection cycle,
+      // which then causes a changed after checked error if the input element was focused before.
+      element.blur();
+    }
+  }
+  private _disabled: boolean;
+
+  /** Emits when a `change` event is fired on this `<input>`. */
+  @Output('cdkDatepickerChange')
+  readonly change: EventEmitter<CdkDatepickerInputEvent<D>> =
+      new EventEmitter<CdkDatepickerInputEvent<D>>();
+
+  /** Emits when an `input` event is fired on this `<input>`. */
+  @Output('cdkDatepickerInput')
+  readonly input: EventEmitter<CdkDatepickerInputEvent<D>> =
+      new EventEmitter<CdkDatepickerInputEvent<D>>();
+
+  /** Emits when the value changes (either due to user input or programmatic change). */
+  _valueChange = new EventEmitter<D | null>();
+
+  /** Emits when the disabled state has changed. */
+  _disabledChange = new EventEmitter<boolean>();
+
+  /** Implemented as part of ControlValueAccessor. */
+  _onTouched = () => {};
+
+  /** Implemented as part of ControlValueAccessor. */
+  protected _controlValAccOnChange: (value: any) => void = () => {};
```

Comment:
Does this need to be accessed in the subclasses? if so we should give it a non-underscore name because its part of the API we expose to users, if not we should make it private

Reply:
Ok, so let's remove the underscore and also give it a name with less abbreviations `controlValueAccessorOnChange`